lib/aws-record/record/attributes.rb

```diff
@@ -382,6 +382,77 @@ def numeric_set_attr(name, opts = {})
           attr(name, Marshalers::NumericSetMarshaler.new(opts), opts)
         end
 
+        # Define an atomic counter attribute for your model.
+        #
+        # Atomic counter are an integer-type attribute that is incremented,
+        # unconditionally, without interfering with other write requests.
+        # The numeric value increments each time you call +increment_<attr>!+.
+        # If a specific numeric value are passed in the call, the attribute will
+        # increment by that value.
+        #
+        # To use +increment_<attr>!+ method, the following condition must be true:
+        # * None of the attributes have dirty changes.
+        # * If there is a value passed in, it must be an integer.
+        # For more information, see
+        # {https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/WorkingWithItems.html#WorkingWithItems.AtomicCounters Atomic counter}
+        # in the Amazon DynamoDB Developer Guide.
+        #
+        # @param [Symbol] name Name of this attribute.  It should be a name that
+        #   is safe to use as a method.
+        # @param [Hash] opts
+        # @option opts [Object] :default_value Optional attribute used to
+        #   define a "default value" to be used if the attribute's value on an
+        #   item is nil or not set at persistence time. The "default value" of
+        #   the attribute starts at 0.
+        #
+        # @example Usage Example
+        #   class MyRecord
+        #     include Aws::Record
+        #     integer_attr :id, hash_key: true
+        #     atomic_counter :counter
+        #   end
+        #
+        #   record = MyRecord.new(id: 1)
```

Comment:
This example technically wouldn't work, because the record would be dirty.  Maybe use `find` here instead.

Reply:
That is correct, so I have updated using `find` as you suggested. Thank you!